velox/dwio/common/ColumnVisitors.h

```diff
@@ -1405,14 +1411,33 @@ class ExtractStringDictionaryToGenericHook {
 
  private:
   int32_t dictionarySize() const {
-    return state_.dictionary.numValues;
+    return state_->dictionary.numValues;
   }
 
   ValueHook* const hook_;
   RowSet const rows_;
-  RawScanState state_;
+  RawScanState* state_;
 };
 
+template <>
+inline void ExtractStringDictionaryToGenericHook::addValue<int32_t>(
```

Comment:
Now this can go back because it won't be compiled with anything other than `V = int32_t`

Reply:
I tried to roll back this code to the previous logic, but an exception occurred during compilation, so I think more work needs to be done.
```
/data/home/wypb/data/code/apache/velox/./velox/dwio/common/ColumnVisitors.h:517:57: error: cannot convert ‘folly::Range<const char*>’ to ‘int32_t’ {aka ‘int’}
  517 |     values_.addValue(rowIndex_ + reader_->numScanned(), value);
      |                                                         ^~~~~
      |                                                         |
      |                                                         folly::Range<const char*>
/data/home/wypb/data/code/apache/velox/./velox/dwio/common/ColumnVisitors.h:1395:15: note:   initializing argument 2 of ‘void facebook::velox::dwio::common::ExtractStringDictionaryToGenericHook::addValue(facebook::velox::vector_size_t, int32_t)’
 1395 |       int32_t value) {
      |       ~~~~~~~~^~~~~
/data/home/wypb/data/code/apache/velox/./velox/dwio/common/ColumnVisitors.h:519:33: error: cannot convert ‘folly::Range<const char*>’ to ‘int32_t’ {aka ‘int’}
  519 |     values_.addValue(rowIndex_, value);
      |                                 ^~~~~
      |                                 |
      |                                 folly::Range<const char*>
```